apps/meteor/app/livechat/server/lib/LivechatTyped.ts

```diff
@@ -363,6 +375,10 @@ class LivechatClass {
 			}
 		}
 
+		if (await LivechatContacts.isChannelBlocked(this.makeVisitorAssociation(visitor._id, roomInfo.source))) {
```

Comment:
Think this should happen before trying to switch visitor deps (L366). If the visitor is blocked, we shouldn't do any action on them.